### ISSUE TITLE
ArrayIndexOutOfBoundsException with word2vec

### ISSUE THREAD

**Issue by eilonreshef** [deeplearning4j/deeplearning4j#1553]
Running a word2vec process almost as-is based on the web documentation. When setting the window size to 7, getting ArrayIndexOutOfBoundsException:

        int batchSize = 1000;
        int iterations = 3;
        int layerSize = 150;

        Word2Vec vec = new Word2Vec.Builder()
                .batchSize(batchSize) //# words per mini-batch.
                .minWordFrequency(5) //
                .useAdaGrad(false) //
                .layerSize(layerSize) // word feature vector size
                .iterations(iterations) // # iterations to train
                .learningRate(0.025) //
                .minLearningRate(1e-3) // learning rate decays wrt # words. floor learning
                .negativeSample(10) // sample size 10 words
                .iterate(iter) //
                .tokenizerFactory(tokenizer)
                .windowSize(7)
                .build();
        vec.fit();

Exception in thread "VectorCalculationsThread 3" java.lang.RuntimeException: java.lang.ArrayIndexOutOfBoundsException: 1000
	at org.deeplearning4j.models.sequencevectors.SequenceVectors$VectorCalculationsThread.run(SequenceVectors.java:780)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1000
	at org.deeplearning4j.models.embeddings.learning.impl.elements.SkipGram.iterateSample(SkipGram.java:211)
	at org.deeplearning4j.models.embeddings.learning.impl.elements.SkipGram.skipGram(SkipGram.java:124)
	at org.deeplearning4j.models.embeddings.learning.impl.elements.SkipGram.learnSequence(SkipGram.java:99)
	at org.deeplearning4j.models.sequencevectors.SequenceVectors.trainSequence(SequenceVectors.java:153)
	at org.deeplearning4j.models.sequencevectors.SequenceVectors$VectorCalculationsThread.run(SequenceVectors.java:768)

**Comment by raver119** [comment]
This issue was already fixed.

As temporary workaround for rc3.8 - just disable negative sampling.
